### PR TITLE
[gh-20][API] As a user, I can get the list of keywords and their results uploaded by CSV files

### LINKED ISSUE #20: [API] As a user, I can get the list of keywords and their results uploaded by CSV files
## Why

User needs to use the viewing results list function as an API for calling from other client-side like mobile.

## Who Benefits?

The user who wants to use our application functions from other client-side application

### LINKED ISSUE #22: [API] As a user, I can search across all reports by searching-keyword
## Why

User needs to use the viewing results list function across all reports as an API for calling from other client-side like mobile.

## Who Benefits?

The user who wants to use our application functions from other client-side application

### PR DESCRIPTION
**Pull request by nvminhtue**
close #20 #22 

## What happened

API part of getting keyword list 

## Insight

- Add guardian handler for authentication (Error handling included)
- Add API controller for getting keyword list
- Add API controller for getting list by keyword
 
## Proof Of Work

Success case
image

Success in searching by keyword
image

Success in searching by NOT FOUND keyword
image

Failure case
image

### CODE REVIEW

File: test/elixir_internal_certificate_web/controllers/api/v1/user_search_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                   "attributes" => %{
+                     "keyword" => "keyword 2",
+                     "status" => "pending"
+                   },
+                   "id" => _,
+                   "relationships" => %{},
+                   "type" => "user_search"
+                 }
+               ],
+               "included" => []
```
Comment (andyduong1920) [inline test/elixir_internal_certificate_web/controllers/api/v1/user_search_controller_test.exs]: should we test for the `meta` part as well

Comment (nvminhtue) [reply test/elixir_internal_certificate_web/controllers/api/v1/user_search_controller_test.exs]: Updated in [3106803](https://github.com/nvminhtue/elixir_internal_certificate/pull/44/commits/3106803ed2ee151f54bb53ae2623eff0fe12990b)

File: lib/elixir_internal_certificate_web/controllers/api/auth_error_handler.ex [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  alias ElixirInternalCertificateWeb.Api.ErrorView
+
+  def auth_error(conn, {type, _reason}, _opts) do
+    conn
+    |> put_status(:unauthorized)
+    |> put_view(ErrorView)
+    |> render("error.json", %{
+      code: "unauthorized",
+      detail: to_string(type)
```
Comment (andyduong1920) [inline lib/elixir_internal_certificate_web/controllers/api/auth_error_handler.ex]: How about we just simply the message here 

```suggestion
      detail: "Login required"
```

**Review by andyduong1920 — APPROVED** [review]
Just a small comment

File: lib/elixir_internal_certificate_web/controllers/api/auth_error_handler.ex (hunk 2 above)
Comment (nvminhtue) [reply lib/elixir_internal_certificate_web/controllers/api/auth_error_handler.ex]: Updated in [1c3cf1a](https://github.com/nvminhtue/elixir_internal_certificate/pull/44/commits/1c3cf1ac992081b443e4ab1840f95cb2fbce3f0b)

**Review by longnd — APPROVED** [review]

**Review by longnd — APPROVED** [review]

**Pull request merged**